Update Repo

mishyy opened the pull request:
Update your Pex repo to be equal with the actual Pex repo.

zack-harrison commented:
We use a different branch.

See:
https://github.com/LuckyPrison/PermissionsEx/tree/1.23.x

zml2008 committed:
Revert "sponge: Update for API support of permission weights"

This reverts commit ca2944387bb3bb2cdf9d456a32f221151c4bf642.
This change has been postponed until API 8.

zml2008 committed:
core: Automatically reload file backend configuration

zml2008 committed:
Tidy up Sponge impl, take out associated object caching for now due to issues

zml2008 committed:
Don't lazily fetch data stores since it doesn't work for some reason?

zml2008 committed:
Bump kotlin release, update to official formatting

tidy up some other things as well.

zml2008 committed:
Add groupmanager and first pass of luckperms backend

There are the first pieces of a new more automated conversions system
coming in here, but for now these backends have to be configured like
any other PEX backend.

zml2008 committed:
Update to JUnit 5 and remove gettext dependency

SQL tests can no longer be parameterized based on a system property --
this may be restored in the future if it's actually used by anyone.

zml2008 committed:
Allow listening to files that don't exist yet

Fixes issues on first run
Fixes #3035

zml2008 committed:
Fix issues with LuckPerms backend

Known remaining issues:
- rank ladders are not properly converted
- sql is not supported
- multiple affixes are not handled

zml2008 committed:
Move the project to Gradle

As a bonus, this is as good a time as any to fix up some issues with
shading -- now things that shouldn't be included are actually not
shaded, and everything that should be relocated is relocated.

There are still some clunky areas of the build, but everything is
functional in the new system (yay)

zml2008 committed:
Add support for Velocity and Bungee/Waterfall proxy implementations

zml2008 committed:
Nothing to see here, move along

zml2008 committed:
Add initial Fabric implementation

Only has extremely rudimentary hooking for players -- most server
operations are not routed through permissions, but it's something to get
started.

Currently requires a customized variant of loom to build pending PR
acceptance.

IntelliJ doesn't like our usage of the RemapJarTask, so that section
might have to be commented out to correctly import this project.

there's a long way to go...

zml2008 committed:
Switch data listener architecture over to using Consumer rather than own private class

zml2008 committed:
Add support for registering update listeners for a CalculatedSubject

zml2008 committed:
Cache active contexts for subjects

This caching is a somewhat generic system that makes it easy to hold
active contexts in cache for the desired amount of time.

zml2008 committed:
Provide much more thorough integration in fabric

This aims to replace every single operator check made with permission
checks, and to represent any possible command executor as a permission
subject.

There are a few outstanding areas that are incomplete, and this will
need thorough testing since any issues with our checks could result in
security vulnerabilities.

zml2008 committed:
Update travis file to match current build system

zml2008 committed:
Let's try this again, why don't we

zml2008 committed:
And third try maybe?

zml2008 committed:
another try?

zml2008 committed:
Move from squirrelid to per-implementation UUID resolving

Also provide a rudimentary resolver for implementations that don't
provide their own.

This removes two dependencies, plus takes the last usage of jsr305
annotations out of the class path.

zml2008 committed:
Build system tweaks

Automatically verify license headers, and add a task to collect
distributable jars and place them in the root project's build folder.

zml2008 committed:
Correct some incorrectly applied license headers

zml2008 committed:
Allow context parse errors, add IpSet context

IpSet context is implemented across all platforms

zml2008 committed:
bungee: Fix misunderstanding of plugin API

Turns out we have to register our own listeners.

Also since Bungee carries an older version of Guava, we can't use
ImmutableSet.toImmutableSet()

zml2008 committed:
Allow passing a prefix to our loggers

This is used on Fabric to identify our log messages

zml2008 committed:
fabric: Provide a WorldEdit permissions resolver

Hopefully they package the resolver once PEX stabilizes, but for now we
have it. As a bonus, this is the closest thing to a permissions API that
fabric has!

zml2008 committed:
Update links in documentation and build file

Adds info on how to build with gradle to README
New project property buildExclude allows us to exclude fabric builds on
the jenkins until Loom is fixed

zml2008 committed:
Add fallback context handling to increase interoperability between PEX and Sponge contexts

zml2008 committed:
Bump fabric-loom version so public builds will work

zml2008 committed:
Update for MC 1.15 and kotlin 1.3.61

zml2008 committed:
core: Fix import of rank ladders from other backends

zml2008 committed:
core: Implement, document, and test fallback subjects

These subjects fill a hole in compatibility by more closely representing
what the default group used to be in PEX 1.x

zml2008 committed:
Update readme to list new platforms supported by PEX

zml2008 committed:
core: Don't try to cache contexts too early

We now wait until the subject is fully initialized.
This resolves a crash on login for servers using the Bukkit
implementation.

zml2008 committed:
build: allocate less ram
Maybe this will make CI more willing to build the project

zml2008 committed:
core: Don't check same context combination twice

This could happen when the same context combination is defined in both
persistent and transient data. Normally this shouldn't have caused
issues with permissions checks, but did result in parents being listed
twice.

zml2008 committed:
Update documentation for Sponge and Fabric impl details

zml2008 committed:
bukkit: Add WorldGuard region context

zml2008 committed:
bukkit: Handle contexts from Vault queries better

This is still not entirely correct, since querying permissions in a
world will still carry data from a player's current state, but should
give better results, and will give the exact permissions results
expected if a player is queried in their current world.

zml2008 committed:
doc: Document Bukkit metapermissions and Vault quirks

zml2008 committed:
core: Handle per-platform configurations

This allows providing platform-specific options for certain
implementations.

zml2008 committed:
bukkit: Alow a fallback to op level check

This fallback will only happen if the permission is unset (value 0) in
PEX, and is configurable in a new platform-specific configuration
section.

The fallback happens through superperms and Vault checks, but not
through the native API -- the aim is to maintain expected results from
existing APIs without polluting PEX's core with knowledge of operator
status.

Closes #3016

zml2008 committed:
fabric: Bump version to 1.15.1 and disable maven publication

zml2008 committed:
fabric: Remove partial work

shouldn't have been committed in the first place

zml2008 committed:
Allow getting different levels of base directories

This change is for the coming ops list backend that needs to look for an
`ops.json` file in the server root, which can now be provided by the
implementation.

zml2008 committed:
Add data store to import from vanilla MC ops list

zml2008 committed:
core: Implement more automated conversion

Update the pex import command to use the automatically detected import
possibilities
Add a new pex version command to provide more details about the
currently running instance of PEX
Fix a few localized text sending issues while we're here too!

zml2008 committed:
Add uuid argument type

zml2008 committed:
bukkit: Correct Vault world choice

When none is specified in methods, all actions should default to the
player's world when available.

Ichbinjoe committed:
Increase flexibility of date based context parsing

Also converts datetime based context from LocalDateTime to
ZonedDateTime which holds the timezone with the DateTime

Closes #3071

zml2008 committed:
Add callback functionality to text abstraction

Including compatibility for platforms that don't provide native callback
support.

zml2008 committed:
Bump gradle to v6.1

zml2008 committed:
fabric: Implement PEX command sender as a mixin

This'll save just a bit of object creation, and could help with
interoperability in the future.

zml2008 committed:
bukkit, bungee, velocity: Fix command registration issue

zml2008 committed:
bukkit: Fix op status fallback to actually be fallback

as opposed to an override :)

zml2008 committed:
Disable configuration watcher tests until I have time to fix them

zml2008 committed:
First pass at discord embeds and a logo

zml2008 committed:
bukkit: In WG hook, check that we have the right classes

i.e. that we are actually working with WG 7+

zml2008 committed:
core/sql: Correct legacy version detection

zml2008 committed:
Update jenkins to point to new location

zml2008 committed:
core/commands: Correctly set ladder name in /demote

zml2008 committed:
commands: Refactor message formatting for extensions

This reduces the nesting level of some of the message formatting code,
making it a good bit more readable.

This also involves converting all comand implementations to Kotlin, and
begins the journey towards a new commands system.

zml2008 committed:
core: Fix improper overriding of compiler flags

zml2008 committed:
GitBook: [master] 19 pages modified

zml2008 committed:
Redesign localization to work without gettext, and update project to use new system

zml2008 committed:
doc: Update structure for GitBook and localization

zml2008 committed:
doc: Update discord embed to point to new docs

zml2008 committed:
doc: Correct homepage URL

menelmir committed:
Fixes some MySQL tables deployement issues due to key length

Closes #3079

zml2008 committed:
Begin fully migrating to kyori text

This lets us reduce a good bit of duplication and actually focus on interesting things

zml2008 committed:
Work on new commands API

zml2008 committed:
Work on UltraPerms conversion

zml2008 committed:
Continue updating for text

zml2008 committed:
Migrate to common gradle plugin

zml2008 committed:
GitBook: [master] 20 pages modified

zml2008 committed:
finish text changeover

We are now fully on kyori-text

I've removed the WIP commands api -- going to go with a more gradual
transition to a better system rather than stripping everything out all
at once.

zml2008 committed:
bukkit: Handle getLocale move to support 1.8.8 servers

zml2008 committed:
Javadoc and localization tweaks

zml2008 committed:
Fix conversion from PEX v1 configuration files

zml2008 committed:
Update fabric for 1.15.2

zml2008 committed:
Bump gradle to 6.4

zml2008 committed:
First pass of updates for Configurate 3.7

zml2008 committed:
datastore: Locale tweaks

zml2008 committed:
PEX-Bukkit is GPL

zml2008 committed:
Commands phase 2: The commandening

- Break out value-returning arguments into their own CommandElement
  subtype
- Convert a lot of the remaining commands API to Kotlin
- Use a DSL-style builder to create commands in Kotlin

zml2008 committed:
Fix passing null as parameter to translatable components

zml2008 committed:
build: use opinionated 3.0, format with ktlint

zml2008 committed:
Revamp glob parser -- works properly now!

zml2008 committed:
Bump gradle, fix imports

zml2008 committed:
Log uncaptured permissions checks

zml2008 committed:
fabric: Update for MC 1.16-rc1

This probably won't entirely run yet -- at least not until an updated
text-adapter-fabric is pushed.

zml2008 committed:
fix formatting

zml2008 committed:
Reorganize for split source sets

zml2008 committed:
Improve structure, fix a whole bunch of dedicated server issues

zml2008 committed:
Update to 1.16-compatible text adapter

zml2008 committed:
Exclude kotlinx-coroutines from shading

zml2008 committed:
Exclude configurate from shadow jar

zml2008 committed:
Don't include() fabric-language-kotlin anymore

Update fabric.mod.json for more accurate dep specifications.

zml2008 committed:
a touch of tidying

zml2008 committed:
bukkit: Port a bunch of the not-super-hacky bits to Kotlin

zml2008 committed:
Tidy up some warnings and deprecations

zml2008 committed:
Rearrange submodules for nicer folder structure

zml2008 committed:
build: Bump dependency versions

zml2008 committed:
core: Move SubjectTypeDefinition to Kotlin

zml2008 committed:
fabric: 1.16.2

zml2008 committed:
Build with Kotlin 1.4.0

zml2008 committed:
fabric/velocity: Bump dep versions

zml2008 committed:
remove usage of old text extensions

zml2008 committed:
sponge: Fully convert to kotlin

zml2008 committed:
Stop building on JDK12

zml2008 committed:
sponge + bukkit: Fix nullability issues

zml2008 committed:
core: Correctly display booleans in command output

zml2008 committed:
move to adventure: it compiles

zml2008 committed:
build: Bump loom version

zml2008 committed:
sponge: begin update to api8

zml2008 committed:
all: Use pull-based command registration, not push-based

zml2008 committed:
sponge: fixes

zml2008 committed:
sponge: Update for engine-scoped services

zml2008 committed:
fabric: bump to 1.16.3

zml2008 committed:
sponge: a bit of tidying

zml2008 committed:
update for adventure deprecations

zml2008 committed:
core + fabric: Introduce the concept of a fallback permission value

zml2008 committed:
Bump for adventure changes

zml2008 committed:
Update for Configurate 4

zml2008 committed:
Re-import SpongeAPI 7 backend

zml2008 committed:
bulid: Move to GH Actions

zml2008 committed:
alL: update dependencies

zml2008 committed:
sponge: Fix usages of locally-modified API

zml2008 committed:
fabric: 1.16.4

zml2008 committed:
build: Make the task to collect shaded jars more idiomatic

zml2008 committed:
build: Archive artifacts from actions runs

zml2008 committed:
api: set up submodule

zml2008 committed:
build: Produce nicer names for final jars

Clean up older output too.

zml2008 committed:
doc: Correct source links in README

Fixes #3099

zml2008 committed:
*: Begin breaking things out into separate modules

zml2008 committed:
proxy-common, hikary: make thise java, not kotlin

zml2008 committed:
build: Enforce compiling with at least JDK 11

zml2008 committed:
*: Rename kotlin files to UpperCamel

zml2008 committed:
Use release flag instead of source/target options

zml2008 committed:
build: Fix building when >JDK11 (temporarily)

zml2008 committed:
sponge: Update for api changes

zml2008 committed:
core, datastore: Move discovery to a ServiceLoader-based architecture

zml2008 committed:
Move most things into api.

This compiles, but probably has many issues, and will change a lot.

zml2008 committed:
core: Actually publish jars

zml2008 committed:
build: Disable Actions caching on publish job -- it just takes time

zml2008 committed:
*: Update for configurate 4 release

zml2008 committed:
*: Update build configuration to be less... weird?

zml2008 committed:
build: Bump opinionated to 4.0.1

zml2008 committed:
build: Disable experimental warning

zml2008 committed:
sponge: Update plugin descriptor for new resource locations

zml2008 committed:
*: Add errorprone checking to build

renovate-bot committed:
Add renovate.json

zml2008 committed:
Merge pull request #3096 from PEXPlugins/renovate/configure

Configure Renovate

renovate[bot] committed:
chore(deps): update actions/cache action to v2 (#3102)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency gradle to v6.7.1 (#3101)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

zml2008 committed:
velocity: Bump velocity-api to 1.1.2

renovate[bot] committed:
chore(deps): update dependency com.zaxxer:hikaricp to v3.4.5 (#3105)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency net.fabricmc:fabric-loader to v0.10.8 (#3107)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency org.jdbi:jdbi3-bom to v3.17.0 (#3109)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

zml2008 committed:
build: Bump plugin versions

zml2008 committed:
minecraft: Move commands over to cloud library

This brings brigadier compatibility and a nice help menu

renovate[bot] committed:
chore(deps): update dependency org.antlr:antlr4 to v4.9 (#3116)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency org.antlr:antlr4-runtime to v4.9 (#3121)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency org.mockito:mockito-core to v3.6.28 (#3120)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency org.mariadb.jdbc:mariadb-java-client to v2.7.1 (#3119)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency org.jdbi:jdbi3-bom to v3.18.0 (#3124)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

zml2008 committed:
sponge: Update for API changes

zml2008 committed:
api + core + sql: Extract more things into API, make separate module for sql store

zml2008 committed:
*: begin switching to type-safe subject type registration

zml2008 committed:
fabric: Update naming to modern standards

It's not 2015 anymore!

zml2008 committed:
*: correctly format imports

zml2008 committed:
*: Translation system now generates java, not kotlin, and is exposed in API

zml2008 committed:
core: Move context definitions, ip sets, caching values, and supplier serializer back to the java side

Some test tidying too.

zml2008 committed:
fabric: Allow passing a subject override to ServerCommandSource

zml2008 committed:
*: some cleanup and de-kotlining

zml2008 committed:
api: Some renames

zml2008 committed:
api: Another way to create context definitions

zml2008 committed:
api: Add ability to query any value with prefix for node tree

zml2008 committed:
sponge: Update for SpongeAPI changes

zml2008 committed:
fabric: allow multiple game lifetimes of the server running

zml2008 committed:
fabric: Track permission state for function execution

The lifecycle has to change a bit so we can be initialized early enough
for function loads.

zml2008 committed:
fabric: Start on separating out what is API from what isn't

zml2008 committed:
core: Move all into impl subpackage to distinguish from API

zml2008 committed:
platform/*: Tighten visibility of classes where possible

zml2008 committed:
build: Migrate to our own repository that proxies things

zml2008 committed:
fabric: Switch to using some more API over our own mixins

zml2008 committed:
*: Work on moving to a dedicated subject identifier type

zml2008 committed:
add nix config

zml2008 committed:
Some documentation reorganization

zml2008 committed:
fabric: Implement permissions API

zml2008 committed:
build: Bump gradle to 6.8

zml2008 committed:
core, minecraft: Finish first pass commands implementation

All previously supported commands are now implemented.
The Kotlin plugin has been removed from the core implementation now that no Kotlin sources are present.

zml2008 committed:
build: Temporarily disable publishing while API is very in flux

zml2008 committed:
velocity: No more kotlin

zml2008 committed:
fabric: Bump loader verison

zml2008 committed:
bungee: No more kotlin

zml2008 committed:
minecraft: Drop almost entirely unused project reactor dependency

zml2008 committed:
sponge, sponge7: Fix shading relocations

zml2008 committed:
datastore: Clean up initialization process

zml2008 committed:
bukkit: Goodbye kotlin

zml2008 committed:
core: Move some methods only relevant for data stores into DataStoreContext from PermissionsEngine

renovate[bot] committed:
chore(deps): update dependency org.mockito:mockito-core to v3.7.0 (#3152)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency org.antlr:antlr4 to v4.9.1 (#3148)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency com.velocitypowered:velocity-api to v1.1.3 (#3146)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency com.github.ben-manes.caffeine:caffeine to v2.8.8 (#3141)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

zml2008 committed:
api, core: Make default and fallback subject types type-safe

zml2008 committed:
bukkit: Remove unneeded file

zml2008 committed:
build: Bump dependencies

renovate[bot] committed:
chore(deps): update dependency ca.stellardrift:confabricate to v2.0.2 (#3153)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

zml2008 committed:
sponge7: Patch things up to be mostly working

zml2008 committed:
platform/*: Add run tasks to platforms still lacking

zml2008 committed:
minecraft: Command output improvements

zml2008 committed:
bukkit: Add command to force-resend command tree

zml2008 committed:
build: Bump dependencies and resolve warnings

zml2008 committed:
build: Validate that all our artifacts relocate their shaded dependencies

zml2008 committed:
build: Make run tasks consistent across platforms

zml2008 committed:
fabric: Remove specific WorldEdit dependency

WE now directly implements the permissions API we support.

zml2008 committed:
api: Move to an engine builder rather than needing engine access directly

zml2008 committed:
build: Tidy up run task configurations

zml2008 committed:
sponge: Goodbye kotlin

zml2008 committed:
minecraft: Update to cloud 1.4.0

zml2008 committed:
minecraft, platform: Support platform-specific configs again

zml2008 committed:
bukkit: Fix metapermission handling

zml2008 committed:
api, minecraft: a bit of tidying

zml2008 committed:
fabric: Drop kotlin dependency

zml2008 committed:
bukkit: A touch of buildscript tidying

zml2008 committed:
Add command support to fabric, other misc improvement

renovate[bot] committed:
chore(deps): update dependency org.mockito:mockito-core to v3.7.7 (#3162)

dunno if this is still used, but here goes i guess

Co-authored-by: Renovate Bot <bot@renovateapp.com>

zml2008 committed:
minecraft: add `/pex <type> <ident> parent clear` command

renovate[bot] committed:
chore(deps): update dependency gradle to v6.8.1 (#3164)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

zml2008 committed:
sponge: Update for API changes

zml2008 committed:
build: Bump dependency versions

renovate[bot] committed:
chore(deps): update dependency org.postgresql:postgresql to v42.2.19 (#3188)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency com.zaxxer:hikaricp to v4.0.2 (#3185)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency com.github.ben-manes.caffeine:caffeine to v2.9.0 (#3183)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency com.velocitypowered:velocity-api to v1.1.4 (#3184)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency net.minecraftforge:forgeflower to v1.5.478.19 (#3186)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency gradle to v6.8.3 (#3191)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency org.mockito:mockito-core to v3.8.0 (#3192)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency net.fabricmc:fabric-loader to v0.11.2 (#3194)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

zml2008 committed:
build: Update plugin version

zml2008 committed:
fabric: Bump dependency versions

renovate[bot] committed:
chore(deps): update dependency com.zaxxer:hikaricp to v4.0.3 (#3199)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency com.github.ben-manes.versions:com.github.ben-manes.versions.gradle.plugin to v0.38.0 (#3198)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

zml2008 committed:
sponge: Update for API changes

zml2008 committed:
bukkit: Allow checkerframework to be shaded

zml2008 committed:
chore: Bump a variety of dependency versions

renovate[bot] committed:
chore(deps): update dependency net.minecraftforge:forgeflower to v1.5.498.2 (#3182)

Co-authored-by: Renovate Bot <bot@renovateapp.com>
